src/cart/crt_context.c

```diff
@@ -418,8 +425,9 @@ crt_ctx_epi_abort(d_list_t *rlink, void *arg)
 
 	/* abort RPCs in waitq */
 	msg_logged = false;
-	d_list_for_each_entry_safe(rpc_priv, rpc_next, &epi->epi_req_waitq,
-				   crp_epi_link) {
+
+
+	d_list_for_each_entry(rpc_priv, &epi->epi_req_waitq, crp_epi_link) {
```

Comment:
this change looks incorrect to me, if it is possible to delete/free item during traversing the list, need to use d_list_for_each_entry_safe.

482 #define d_list_for_each_entry(pos, head, member)                          \                         
483 >       for (pos = d_list_entry((head)->next, __typeof__(*pos), member),  \                         
484 >       >            prefetch(pos->member.next);                             \                      
485 >            &pos->member != (head);                                         \                      
486 >            pos = d_list_entry(pos->member.next, __typeof__(*pos), member),\                       
487 >            prefetch(pos->member.next))  

If the item being deleted/freed, then the "pos" pointer possibly be invalid the cannot access "pos->member.next".

Reply:
yes, this got fixed in next commit